### DISCUSSION TITLE
Simple_Terminal deprecated?

### DISCUSSION THREAD

**Discussion by jgriffitts** [fltk-rs/fltk-rs#1475] [General]
I see that the Fl_Simple_Terminal widget is being phased out in favor of a new Fl_Terminal widget, first committed on Nov 14.  The Rust wrapper doesn't yet know about Fl_Terminal. Is this on anybody's radar? I'm thinking of submitting a patch to add the new widget, unless somebody else is already working on it.
Ref: https://www.fltk.org/newsgroups.php?s21+gfltk.general+v43042+T0

**Comment by MoAlyousef** [comment]
Hi
I’ve been following the development of Fl_Terminal, but I’m currently very busy since I’m moving out of my apartment in a week. Submitting a patch which adds it would be greatly appreciated.

**Reply by jgriffitts** [reply]
I have put in pull requests with a new wrapper for Fl_Terminal:
[on cfltk repository](https://github.com/MoAlyousef/cfltk/pull/228)
[on fltk-rs repository](https://github.com/fltk-rs/fltk-rs/pull/1478)

These changes are reasonably well tested and completely functional. The Terminal widget wrapper is enclosed in the `experimental` module that was already present in group.rs. This is in group.rs because Terminal is a subclass of Group in the wrapped c++ implementation.

My test program is attached here, for your information:
[main.rs](https://github.com/fltk-rs/fltk-rs/files/13516688/main.rs.txt)

The changes include convenience changes to the Color structure in enums.rs. These changes are optional but I find them useful myself, so please take them or leave them as you choose.

I can see that the Terminal widget is under active development, so I'll try to track it and keep this wrapper up to date.

**Reply by MoAlyousef** [reply]
Thank you. I have merged the PRs. Would you mind if I add your test program to the examples?

**Reply by jgriffitts** [reply]
Please do anything you like with the test program. I assert no control or ownership of it.
